src/transformers/models/mamba/modeling_mamba.py

```diff
@@ -123,14 +125,14 @@ def forward(self, hidden_states: torch.Tensor, inference_params=None):
             )
             hidden_states = hidden_states.unsqueeze(-1)
         else:
-            conv_state = nn.functional.pad(hidden_states, (self.d_conv - hidden_states.shape[-1], 0))
+            conv_state = nn.functional.pad(hidden_states, (self.conv_kernel_size - hidden_states.shape[-1], 0))
             inference_params.conv_states[self.layer_idx].copy_(conv_state)
             hidden_states = causal_conv1d_fn(hidden_states, conv_weights, self.conv1d.bias, activation=self.activation)
```

Comment:
```
model = MambaForCausalLM(config)
import torch
model(torch.tensor([[1,2,3,4,5,6,7,8,9,10]]))
```
![image](https://github.com/huggingface/transformers/assets/53668275/caec4116-4307-4d1a-823c-fc302271af56)

causal_conv1d_fn need var in cuda, so maybe there should add a check?

![image](https://github.com/huggingface/transformers/assets/53668275/ec0d6cb1-ff23-4679-a3eb-341d183b8966)

Reply:
yes we need a check